var/spack/repos/builtin/packages/lammps/package.py

```diff
@@ -854,6 +855,7 @@ def url_for_version(self, version):
     patch("lib.patch", when="@20170901")
     patch("660.patch", when="@20170922")
     patch("gtest_fix.patch", when="@:20210310 %aocc@3.2.0")
+    patch("intel-aocc.patch", when="@20220324: +intel %aocc")
```

Comment:
Since this was merged in upstream on 2022-12-16, it should already be part of the patch release `20221222`. That patch should therefore only be applied until the previous patch release `20221103`.

Reply:
The when condition is still wrong. It should be `@20220324:20221103`. You want it to apply until and including `20221103`.